### PR TITLE
feat(ci): add ci submodule

### PR DESCRIPTION
**Pull request by danielRep**
## PR Description

This PR adds the bao-ci as submodule of the bao-docs repo. This is beneficial to allow the user to use the docker environment locally. 

### Type of change

- **feat**: introduces a new functionality
  - Logical unit: ci

## Checklist:

- [x] The changes follows the documentation guidelines described in [here](https://github.com/bao-project/bao-docs/blob/main/source/development/doc_guidelines.rst).
- [x] The changes generate no new warnings when building the project. If so, I have justified above.
- [x] I have run the CI checkers before submitting the PR to avoid unnecessary runs of the workflow.

### CODE REVIEW

File: .github/workflows/base-ci.yml [hunk 1]
```diff
@@ -29,7 +29,7 @@ jobs:
     container: baoproject/bao:latest
     steps:
       - uses: actions/checkout@v2
-      - run: | 
+      - run: |
```
Comment (josecm) [inline .github/workflows/base-ci.yml:32]: Beware of this kind of change. Removing this kind of trailing white spaces or similar issues should be done in separate commits.

File: .github/workflows/base-ci.yml [hunk 2]
```diff
@@ -38,7 +38,7 @@ jobs:
     container: baoproject/bao:latest
     steps:
       - uses: actions/checkout@v2
-      - run: doc8 source/ --ignore D000
+      - run: make format
```
Comment (josecm) [inline .github/workflows/base-ci.yml:41]: There is already a CI rule called `format` for formatting C files using clang-format. We need another name. Didn't you have issues when adding it to *ci.mk*?

File: .gitmodules [hunk 3]
```diff
@@ -0,0 +1,3 @@
+[submodule "bao-ci"]
+	path = bao-ci
```
Comment (josecm) [inline .gitmodules:2]: The convention stipulated was that the path of the repo would only be `ci`

**Review by josecm — CHANGES_REQUESTED** [review]
Is there any PR in https://github.com/bao-project/bao-ci that adds the *doc8* check to `ci.mk`

**Comment by danielRep** [comment]
> Is there any PR in https://github.com/bao-project/bao-ci that adds the _doc8_ check to `ci.mk`
https://github.com/bao-project/bao-ci/pull/50

**Comment by danielRep** [comment]
> Is there any PR in https://github.com/bao-project/bao-ci that adds the _doc8_ check to `ci.mk`

https://github.com/bao-project/bao-ci/pull/50

File: .gitmodules (hunk 3 above)
Comment (danielRep) [reply .gitmodules]: Fixed [a0ff278](https://github.com/bao-project/bao-docs/pull/68/commits/a0ff27868a276daf8cdd400a82096f2d7e121683)

File: .github/workflows/base-ci.yml (hunk 2 above)
Comment (danielRep) [reply .github/workflows/base-ci.yml]: Fixed. [e134cc0](https://github.com/bao-project/bao-docs/pull/68/commits/e134cc098c330665e4212c86fc15398ae78bf61f)

File: .github/workflows/base-ci.yml [hunk 4]
```diff
@@ -29,20 +29,29 @@ jobs:
     container: baoproject/bao:latest
     steps:
       - uses: actions/checkout@v2
```
Comment (josecm) [inline .github/workflows/base-ci.yml:31]: Just a small note: we must update the checkout action to v4 as v2 is deprecated. But this can be addressed in a later commit that does this for all workflows.

**Review by josecm — APPROVED** [review]

**Review by miguelafsilva5 — APPROVED** [review]
